README.md

```diff
@@ -143,11 +143,10 @@ required for all threaded applications and is very likely to be part of your
 system already, because it comes with libc. Next, install cntlm onto your
 system like so:
 
-Default installation directories are /usr/sbin, /usr/share/man and /etc. Should
-you want to install cntlm into a different location, change the DESTDIR
+Default installation directories are /usr/local/sbin, /usr/local/share/man and /etc.
+Should you want to install cntlm into a different location, change the DESTDIR
```

Comment:
This is still not fully right. There should be a `PREFIX` variable as well with the default set to `/usr/local` that can be overridden to use `/usr` instead for packaging. `BINDIR` and `MANDIR` should use this variable.

Reply:
I added the PREFIX variable as you suggested.

Reply:
Looks great! It guess that some packaging scripts might still need an update. The rpm one seems good as it actually does the overrides the way you had mentioned already, but the deb one isn't clear. Maybe there are others too. Implementing `--prefix` in the configure script might do the trick already for Debian's tooling.

Reply:
Rpm and deb scripts are still working, this change has no impact on them. If you think this pr is ok I can merge it.

Reply:
I don't understand. Why would this not affect the packaging scripts?

Reply:
Because PREFIX is an additional variable that can provide more fine grained configuration options, but scripts can ignore it and still use the others BINDIR, MANDIR, SYSCONFDIR, DESTDIR.

Reply:
Yes but this PR changes the default right? So if scripts were relying on the default being `/usr` then now they would end up with `/usr/local` instead. In that case the scripts need adjustment to set the new PREFIX variable to `/usr` again.

Reply:
Yes, you mean `DAEMON=/usr/sbin/cntlm` in `cntlm.init` and `cntlm.sysconfig` files? I'd rather change these defaults, too. All the point of this pr is that /usr/local is a better position than /usr, so it is better to change these defaults accordingly.

Reply:
No, that's not the case. The PREFIX for the packaged versions must be `/usr`. Only manually installed versions should end up in `/usr/local`. These init scripts are generally used by the packages and thus should stay with `/usr`. The other option would be to generate the right paths in them.

Reply:
See here for instance: https://lintian.debian.org/tags/file-in-usr-local

Reply:
No, the default for any system should be `/usr/local`. But packaging scripts needs to end up putting the files in `/usr`. Both should be supported. So this PR is a good thing and the RPM scripts already seem to cope with this, but we should just check and possibly fix the DEB ones.

Reply:
Finally I see what you mean. This line

`$(MAKE) DESTDIR=$(CURDIR)/debian/cntlm install`

in the `rules/deb` file.
This must be changed with

`$(MAKE) DESTDIR=$(CURDIR)/debian/cntlm PREFIX=/usr install`

Do you agree? is there anything else?

Reply:
Yes, exactly! :) Possibly it could also be good to add it to the line with just `$(MAKE)` a bit higher,  but I think this would only really change things if libraries are build from the same build.

(And I don't know if there are more packaging scripts that would need a change.)

Reply:
> Yes, exactly! :) Possibly it could also be good to add it to the line with just `$(MAKE)` a bit higher

In that line the variable is not used, since it is relevant only for the install phase, while `make` alone just builds the executable.

Reply:
Yes indeed. As far as I can tell because it's just one executable rpath doesn't come into play. In case it does, it needs to be added there too. (That's why usually you do see it in other builds.)
https://en.wikipedia.org/wiki/Rpath

Reply:
If it's ok for you I can merge this PR.

Reply:
Looks great to me!!